onnxruntime/core/optimizer/nhwc_transformer.cc

```diff
@@ -237,6 +249,35 @@ void NhwcTransformerImpl::TransformQLinearGlobalAveragePool(Node& node) {
   CreateNhwcArgument(node, node, nhwc_input->rank_);
 }
 
+void NhwcTransformerImpl::TransformSplit(Node& node) {
+  auto& input_defs = node.MutableInputDefs();
+
+  auto* nhwc_input = LookupNhwcArgument(input_defs[0]);
+  if (nhwc_input == nullptr) {
+    return;
+  }
+
+  // Change the axis attribute accordingly for NCHW to NHWC model.
+  const auto* axis_attr = graph_utils::GetNodeAttribute(node, "axis");
+  if (axis_attr != nullptr && utils::HasInt(*axis_attr)) {
+    int64_t axis = axis_attr->i();
+    if (axis < -nhwc_input->rank_ || axis >= nhwc_input->rank_) {
+      // direct return on invalid axis
+      return;
+    }
+    if (axis >= 1) {
```

Comment:
I think it might be easier to read this code to have something more like what is in HandleNegativeAxis.

```
if (axis < 0) {
  axis = axis + nhwc_input->rank_;
}
if (axis == 1) {
  axis = nhwc_input_rank_->rank_ - 1;
} else if (axis > 1) {
  axis = axis - 1;
}
```

Reply:
Yeah.